libs/zglfw/build.zig

```diff
@@ -3,12 +3,15 @@ const std = @import("std");
 pub const Package = struct {
     zglfw: *std.Build.Module,
     zglfw_c_cpp: *std.Build.CompileStep,
+    options: Options,
 
     pub fn link(pkg: Package, exe: *std.Build.CompileStep) void {
         exe.addModule("zglfw", pkg.zglfw);
 
         const host = (std.zig.system.NativeTargetInfo.detect(exe.target) catch unreachable).target;
 
+        if (pkg.options.emscripten) return;
+
         switch (host.os.tag) {
             .windows => {},
             .macos => {
```

Comment:
We can just leverage the Zigs builtin emscripten target here and treat it like the others.

Reply:
Just commited this. Not sure if its better. Its still tricky because zig modules reuse target from target its linked with (as I understand), now zglfw and zgpu etc. has to assume both .emscripten and .freestanding target means emscripten. It works as long as we don't need any other .freestanding target. But still we have to  support `wasm32-freestanding` and `emscripten` os, or beter workaround has to be found to be able to use `.emscripten` target everywhere.

Reply:
> now zglfw and zgpu etc. has to assume both .emscripten and .freestanding target means emscripten

Why? I don't understand. Do you mean that if we compile our game as freestanding we are unable to link zglfw that has been built with emscripten as os? I thought that it would link because they are both wasm32.

Reply:
Basically issue is that I had to use `wasm32-freestanding` instead of emscripten for main user/sample code, otherwise I get std lib errors such as:
```
zig-master\lib\std\os.zig:153:24: error: root struct of file 'c' has no member named 'fd_t'
```

In regards to zglfw & libs issue is not with `c` but zig modules that don't support specifying different target for them: [build.createModule](https://ziglang.org/documentation/master/std/#A;std:Build.createModule) - can't specify target. [addModule()](https://ziglang.org/documentation/master/std/#A;std:Build.addModule) just uses target from exe its added to. 
So zig bindings code such as `zglfw.zig` in end has to assume that `builtin.target` freestanding is emscripten.
Hope I explained it better this time.

Reply:
Yes, i follow now. Thanks for clarifying. I'm not sure whether it's better to deal with the entry point or not in the long run; but let's go with what works for you for now.